shell/plugins/dashboard-store/mutations.js

```diff
@@ -98,37 +98,52 @@ export function load(state, {
 
   cache.generation++;
 
-  let entry;
+  let entry = cache.map.get(id);
+  const inMap = !!entry;
 
+  //
+  // Determine the `entry` that should be in the local map and list cache
+  //
   if ( existing && !existing.id ) {
-    // A specific proxy instance to used was passed in (for create -> save),
-    // use it instead of making a new proxy
-    entry = replaceResource(existing, data, getters);
-    addObject(cache.list, entry);
```

Comment:
addObject used here (create resource) and below (load new resource via socket) iterates over entire list